src/engine/GroupBy.h

```diff
@@ -244,13 +264,56 @@ class GroupBy : public Operation {
   std::optional<HashMapOptimizationData> checkIfHashMapOptimizationPossible(
       std::vector<Aggregate>& aggregates);
 
+  // Extract values from `expressionResult` and store them in the rows of
+  // `resultTable` specified by the indices in `evaluationContext`, in column
+  // `outCol`.
+  static void extractValues(
+      sparqlExpression::ExpressionResult&& expressionResult,
+      sparqlExpression::EvaluationContext& evaluationContext,
```

Comment:
Do you need the complete `evaluationContext`, or does as single integer (the start index) suffice?

Reply:
We need the full evaluationContext here, because the `makeGenerator` function requires it.